skxray/speckle_visibility/speckle_visibility.py

```diff
@@ -0,0 +1,293 @@
+#                                                                      #
+# * Redistributions in binary form must reproduce the above copyright  #
+#   notice this list of conditions and the following disclaimer in     #
+#   the documentation and/or other materials provided with the         #
+#   distribution.                                                      #
+#                                                                      #
+# * Neither the name of the Brookhaven Science Associates, Brookhaven  #
+#   National Laboratory nor the names of its contributors may be used  #
+#   to endorse or promote products derived from this software without  #
+#   specific prior written permission.                                 #
+#                                                                      #
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS  #
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT    #
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS    #
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE       #
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,           #
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES   #
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR   #
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)   #
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,  #
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+
+"""
+    This module will provide statistics analysis for the speckle pattern
+    to use in X-ray Speckle Visibility Spectroscopy (XSVS)
+"""
+
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+import numpy as np
+from six.moves import zip
+from six import string_types
+
+import skxray.correlation as corr
+import skxray.roi as roi
+import skxray.core as core
+
+try:
+    iteritems = dict.iteritems
+except AttributeError:
+    iteritems = dict.items  # python 3
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def max_counts(sample_dict, label_array):
+    """
+    This will determine the highest speckle counts occurred in the required
+    ROI's in required images.
+
+    Parameters
+    ----------
+    sample_dict : dict
+        sets of images as a dictionary
+
+    label_array : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    max_counts : int
+        maximum speckle counts
+    """
+    max_cts = 0
+    for key, img_sets in iteritems(sample_dict):
+        for n, img in enumerate(img_sets.operands[0]):
```

Comment:
I think http://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.measurements.maximum.html#scipy.ndimage.measurements.maximum implements this

```python
import scipy.ndimage as ndi
max_cts = 0
for img_set in images_sets:
    for image in img_set:
        frame_max = ndi.measurements.maximum(img, label_array)
        max_cts = max(max_cts, frame_max)
return max_cts
```